server/services/auto_tag.py

```
@@ -36,39 +42,79 @@ async def _tag_and_insert(
-            **fields,
-            "user_id": user_id,
-            "image_url": image_url,
-            "is_available": False,
-            "tag_confidence": round(confidence, 3),
-            "tagged_at": datetime.now(timezone.utc).isoformat(),
-        }
+    # Semaphore is acquired by the caller (`_tag_and_insert_with_timeout`) so
+    # queue-wait does not consume the per-image timeout budget.
+    try:
+        fields, confidence = await asyncio.to_thread(tag_image, image_url)
+    except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)